app/models/concerns/test_track/identity.rb

```diff
@@ -46,6 +46,17 @@ def test_track_identifier(identifier_type, identifier_value_method)
             end
           end
         end
+
+        define_method :test_track_sign_up! do
+          discriminator = TestTrack::IdentitySessionDiscriminator.new(self)
+          identifier_value = send(identifier_value_method)
+
+          if discriminator.participate_in_unauthenticated_session?
+            discriminator.controller.send(:test_track_session).sign_up! identifier_type, identifier_value
+          else
+            TestTrack::OfflineSession.create_visitor_for(identifier_type, identifier_value)
```

Comment:
Yea I flip flopped all over this one also. The though was that if a new user was created via some job or as a side effect of another action we would want it to get a visitor and be trackable.

i.e.
```
# Non-web context, maybe a background job
user = User.new(name: 'foo')
user.do_trackable_thing

# Sometime later, actual person comes in to claim that user as there own, in web context
current_user = User.claim(name: 'foo')
current_user.do_other_trackable_thing
```

We'd want `do_trackable_thing` and `do_other_trackable_thing` to link back to the same visitor.  However, I think that we get that anyway per the visitor_lookup code you posted. `do_trackable_thing` would create a new visitor anyway.  

So yea, raise-y sgtm.